src/Microsoft.OData.Core/UriParser/SemanticAst/ODataPathExtensions.cs

```diff
@@ -146,8 +167,14 @@ public static ODataPath TrimEndingKeySegment(this ODataPath path)
         /// <returns>The ODataPath without type-cast in the end</returns>
         public static ODataPath TrimEndingTypeSegment(this ODataPath path)

```

Comment:
Suggestion: You could consider moving all the public methods before all the internal methods.

Reply:
#Resolved